wallet/src/wallet.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::path::Path;
+
+use common::chain::{OutPoint, Transaction, TxOutput};
+use common::primitives::{Id, Idable};
+use utxo::Utxo;
+use wallet_storage::{
+    TransactionRw, Transactional, WalletStorageImpl, WalletStorageTxRwImpl, WalletStorageWrite,
+};
+use wallet_types::wallet_tx::{TxState, WalletTx};
+
+/// Wallet errors
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum WalletError {
+    #[error("Wallet database error: {0}")]
+    DatabaseError(wallet_storage::Error),
+    #[error("Transaction already present: {0}")]
+    DuplicateTransaction(Id<Transaction>),
+}
+#[allow(dead_code)] // TODO remove
+pub struct Wallet {
+    db: WalletStorageImpl,
```

Comment:
A better way to do this is to use a generic, like we do in chainstate storage (see chainstate launcher). The wallet shouldn't only work with sqlite storage. There should be a generic parameter that determintes the storage backend at compile-time.